src/viam/sdk/rpc/dial.cpp

```diff
@@ -56,6 +59,10 @@ void DialOptions::set_entity(boost::optional<std::string> entity) {
     auth_entity_ = entity;
 }
 
+void DialOptions::set_timeout(std::chrono::seconds timeout) {
```

Comment:
Seconds are fairly coarse, and since they are denominated in integers you couldn't, as this is written, express a 2.5 second timeout. I'd consider either:
1) Going to `std::milliseconds` OR
2) Taking `std::chrono::duration<float>` which implicitly represents seconds but can take a non-integer value.

Reply:
Fair point; I chose option 2. At the rust level (and I believe also now in Python), the timeout value is a float representing seconds. I think we want to allow non-integer values, but maintain the unit across the SDKs where possible.